base/applications/charmap/map.c

```diff
@@ -312,64 +312,38 @@ OnClick(PMAP infoPtr,
         WORD ptx,
         WORD pty)
 {
-    POINT pt;
     INT x, y;
 
-    pt.x = ptx;
-    pt.y = pty;
+    x = ptx / infoPtr->CellSize.cx;
+    y = pty / infoPtr->CellSize.cy;
```

Comment:
Are we sure that infoPtr->CellSize.cx and cy will never ever be == 0 ?

Reply:
The cell size is computed on WM_CREATE and is dependent on the client rect.  So cx and cy will be non-zero as long as the client rect is calculated correctly (comes from the rc I believe).  That being said, I can protect against this if you would like, and bail out if either of them are zero.

Reply:
You can manage them to be greater or equal to '1' instead.

Reply:
Do you mean just doing something like this?

`x = ptx / max(1, infoPtr->CellSize.cx);`

Reply:
yes, for example. Or alternatively fixing all the places where the CellSize.cx/cy is initialized.

Reply:
Okay, what I'll probably do is implement as above (using max) for this commit.  Then I'll prepare a new pull request that can protect against CellSize containing 0 and perhaps some other charmap fixes.  Does that sound okay?

Reply:
Sounds good to me!

Reply:
Should be done in the latest commit.  Thanks!